firstvoices/backend/views/custom_order_recalculate_views.py

```diff
@@ -1,51 +1,154 @@
+            403: OpenApiResponse(
+                description="Todo: Action not authorized for this User"
+            ),
+            404: OpenApiResponse(description="Todo: Site not found"),
+        },
+    ),
+    get_preview=extend_schema(
+        description="Returns the most recent custom order recalculation preview results for the specified site. "
+        "Preview results are not saved to the database.",
+        responses={
+            200: CustomOrderRecalculationResultSerializer,
+            403: OpenApiResponse(description="Todo: Not authorized for this Site"),
+            404: OpenApiResponse(description="Todo: Site not found"),
+        },
+    ),
+    create_preview=extend_schema(
+        description="Queues a custom order recalculation preview task for the specified site. "
+        "dictionary-cleanup/preview and dictionary-cleanup/create_preview are the same endpoint.",
+        responses={
+            201: OpenApiResponse(description="Recalculation preview has been queued."),
+            403: OpenApiResponse(
+                description="Todo: Action not authorized for this User"
+            ),
+            404: OpenApiResponse(description="Todo: Site not found"),
+        },
+    ),
+)
+class CustomOrderRecalculateViewset(
+    CustomOrderFVPermissionViewSetMixin,
+    SiteContentViewSetMixin,
+    ListViewOnlyModelViewSet,
 ):
     http_method_names = ["get", "post"]
-    serializer_class = CustomOrderRecalculationPreviewResultDetailSerializer
+    serializer_class = CustomOrderRecalculationResultSerializer
 
     def get_queryset(self):
         site = self.get_validated_site()
         if site.count() > 0:
-            return CustomOrderRecalculationPreviewResult.objects.filter(
-                site__slug=site[0].slug
+            return CustomOrderRecalculationResult.objects.filter(
+                site__slug=site[0].slug, is_preview=False
             ).order_by("-latest_recalculation_date")
         else:
-            return CustomOrderRecalculationPreviewResult.objects.none()
+            return CustomOrderRecalculationResult.objects.none()
 
     def create(self, request, *args, **kwargs):
         site = self.get_validated_site()
         site_slug = site[0].slug
 
+        # Call the recalculation task
+        try:
+            recalculation_results = recalculate_custom_order.apply_async((site_slug,))
+
+            # Delete any previous results
+            CustomOrderRecalculationResult.objects.filter(
+                site=site[0], is_preview=False
+            ).delete()
+
+            # Save the result to the database
+            CustomOrderRecalculationResult.objects.create(
+                site=site[0],
+                latest_recalculation_result=recalculation_results.get(timeout=360),
+                task_id=recalculation_results.task_id,
+                is_preview=False,
+            )
+
+            return Response({"message": "Recalculation has been queued."}, status=201)
```

Comment:
Turns out there is actually a status code for "accepted/queued", so we could use 202 here.